api/src/main/java/ai/djl/training/optimizer/Adam.java

```diff
@@ -46,6 +46,7 @@ public class Adam extends Optimizer {
 
     private Map<String, Map<Device, NDArray>> means;
     private Map<String, Map<Device, NDArray>> variances;
+    private boolean adamw;
```

Comment:
It's probably cleaner to have a separate AdamW class. Especially as it gives a place for separate documentation for it. For simplicity, you can still leave the internal implementations connected and/or have one of them extend the other (or extend a third class)